src/sema/mutability.rs

```diff
@@ -353,11 +333,19 @@ fn recurse_statements(stmts: &[Statement], ns: &Namespace, state: &mut StateChec
                 recurse_yul_statements(&inline_assembly.body, state);
             }
         }
+
+        state
+            .diagnostics
+            .extend(state.per_statement_diagnostics.drain().map(|(_, v)| v));
```

Comment:
Yeah good point. I've analyzed the situation a bit further. There are situations where the same error will be reported twice, if `Expression::StorageLoad` will be considered during expr recursing. If this can be made consistent somehow, there's no need to restrict diagnostics to a context.

Reply:
Just added the edge case for storageloads ond struct members

Reply:
So why is a storageload of a struct member an exception? **Any** storage load should require view mutability.  

The comment just explains how struct member works but nothing about why this is an exception.

This is just an ugly hack, I am not going to approve it.